### DISCUSSION TITLE
NSwag/Swashbuckle issues

### DISCUSSION THREAD

**Discussion by henda79** [Azure/azure-mobile-apps#881] [General]
I started to use Swashbuckle sometime ago but now I see that its not supported on .NET 8, so with that, I decided to try NSwag

I'm struggling to get it setup, on thing I can't work out, is where does UseSwaggerUI3 method come from (this is in the docs for Datasync) ? All I can find is UseSwaggerUi extension method, but with this I get an exception:

System.EntryPointNotFoundException: 'Entry point was not found.'

when AddDatasyncProcessors is executed.

any ideas ?

**Comment by adrianhall** [comment]
I have, unfortunately, not used NSwag + .NET 8.  Having said that, care to publish a quick repro?  I’ll take a look and see what is going on.

**Reply by Arslan007** [reply]
I was having issue to update to .NET8 as well. 
I believe issue is Data sync library depends on this NSwag.AspNetCore 13.20.0 and it depends on other dependencies which doesnot support .NET8. I believe it needs to update to 14.0.2

**Reply by adrianhall** [reply]
I can certainly update dependencies as long as they support .NET6 as well.  Given it’s a major version upgrade, it may be problematic updating it for breaking changes.  Open an issue, and I’ll take a look this week.

**Reply by henda79** [reply]
@Arslan007 , thanks for opening the issue. I would have sorted that but with the time differences and all of that...

@adrianhall , I had Swashbuckle working fine on .NET 8 and wasn't having any issues, but once I found out it was no longer supported / mainitained, I ripped it all out and tried NSwag.

Does anyone have advice on this missing method UseSwaggerUI3/UseSwaggerUi3 ? I see this in various docs including Datasync but can't find it.

**Reply by adrianhall** [reply]
For the SwaggerUI3 thing, see the official docs here: https://github.com/RicoSuter/NSwag/wiki/AspNetCore-Middleware

Basically, Rico Suter is the maintainer of NSwag - Azure Mobile Apps doesn't use SwaggerUI3 - it's something from a 3P library that we have no interaction with.  Thus, you should open an issue there if you are having problems getting UseSwaggerUI3 to work.

NSwag update will be in the 6.1.0 release going out this afternoon.

**Reply by hansmbakker** [reply]
Look-ahead at .NET 9 (you might already have seen this?) - asp.net core is moving to its own OpenAPI libraries:

https://github.com/dotnet/aspnetcore/issues/54599?darkschemeovr=1
